femwell/maxwell/waveguide.py

```diff
@@ -280,9 +282,119 @@ def plot(self, field, plot_vectors=False, colorbar=True, direction="y", title="E
             direction=direction,
         )
 
-    def show(self, field, **kwargs):
-        self.plot(field=field, **kwargs)
-        plt.show()
+    def plot_component(
+        self,
+        field: Literal["E", "H"],
+        component: Literal["x", "y", "z", "n", "t"],
+        part: Literal["real", "imag", "abs"] = "real",
+        boundaries: bool = True,
+        colorbar: bool = False,
+        ax: Axes = None,
+    ):
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+        if part == "real":
+            conv_func = np.real
+        elif part == "imag":
+            conv_func = np.imag
+        elif part == "abs":
+            conv_func = np.abs
+        else:
+            raise ValueError("A valid part is 'real', 'imag' or 'abs'.")
+
+        if ax is None:
+            fig, ax = plt.subplots()
+        else:
+            fig = plt.gcf()
```

Comment:
why do we need fig here? I'd guess plot_component should only care about an `ax` and return that (and generate one if it's `None`
Maybe have a look on skfem https://github.com/kinnala/scikit-fem/blob/580df1af6d744293a71d959ae98ca006876cdfc8/skfem/visuals/matplotlib.py#L202 I'm thinking of keeping this similar

What do you think?

Reply:
I simply copy the way it was made in `plot_intensity()` but the way you proposed was cleaner!